tests/adapters/test_tool.py

```diff
@@ -177,6 +180,13 @@ def test_tool_from_function_with_pydantic_nesting():
     assert tool.args["profile"]["properties"]["contact"]["type"] == "object"
     assert tool.args["profile"]["properties"]["contact"]["properties"]["email"]["type"] == "string"
 
+    # Reference should be resolved for nested pydantic models
```

Comment:
nit: it's totally fine as it is, but since the expected json schema is somehow complex, it might be more readable if we define the entire dict and assert `assert tool.args == expected_schema`